src/python/dxpy/dx_extract_utils/ExpressionInputsValidator.py

```diff
@@ -0,0 +1,179 @@
+import dxpy
+
+from ..exceptions import (
+    err_exit,
+    PermissionDenied,
+    InvalidInput,
+    InvalidState,
+    ResourceNotFound,
+    default_expected_exceptions,
+)
+from ..utils.resolver import resolve_existing_path, is_hashid, ResolutionError
+
+import json
+import os
+
+
+class ExpressionInputsValidator:
+    """InputsValidator class for extract_assay expresion. Checks for invalid input combinations"""
+
+    def __init__(self, args):
+        self.error_handler = None
+
+        self.list_assays = args.list_assays
+        self.retrieve_expression_flag = args.retrieve_expression
+        self.assay_name = args.assay_name
+        self.input_json = args.input_json
+        self.input_json_file = args.input_json_file
+        self.json_help = args.json_help
+        self.sql = args.sql
+        self.additional_fields = args.additional_fields
+        self.additional_fields_help = args.additional_fields_help
+        self.expression_matrix = args.expression_matrix
+        self.delim = args.delim
+        self.output = args.output
+
+        self.validations_to_run()
```

Comment:
`__init__ should be used to only initialize the object's attributes. So I suggest removing this line entirely, and then explicitly calling the `validations_to_run` method on an instantiated object of the class.